core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingManagerImpl.java

```diff
@@ -85,10 +75,11 @@ public AvailabilityMode getAvailabilityMode() {
       return availabilityMode;
    }
 
-   private CompletionStage<Void> updateAvailabilityMode(AvailabilityMode mode) {
-      log.debugf("Updating availability for cache %s: %s -> %s", cacheName, this.availabilityMode, mode);
+   private void updateAvailabilityMode(AvailabilityMode mode) {
+      if (log.isDebugEnabled()) {
```

Comment:
-1, this log is rare enough that they're not going to hurt performance (as are/should be debug logs in general)

Reply:
but you said "is rare enough that they're not going to hurt performance"!
so, is `log.isDebugEnabled()` going impact the performance?
I'll revert it.

Reply:
I wasn't saying to remove it because of performance, but because most debug logs don't have `if (log.isDebugEnabled())` checks (because they don't help). 

A reader of the code might wonder what's so special about this particular debug log to make it worth a check, and the less the reader has to wonder about while trying to understand the code, the better.